nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -146,19 +144,17 @@ LifecycleManager::destroyLifecycleServiceClients()
 bool
 LifecycleManager::createBondConnection(const std::string & node_name)
 {
-  const double timeout_ns =
-    std::chrono::duration_cast<std::chrono::nanoseconds>(bond_timeout_).count();
-  const double timeout_s = timeout_ns / 1e9;
+  const double timeout_s = bond_timeout_.count();
```

Comment:
My understanding is that `count` gives you back out the count of the units, which in this case is now seconds. This would then lose all sig figs below an integer, which could mess up the other things in this function. I think you should revert `bond_timeout_` to a `chrono milliseconds` or `nanoseconds` variable type and convert the `s`-typed parameter into it and update this function appropriately.

I think `timeout_s` would be 0 in alot of cases and the `from_seconds` cast I'm not sure what it will do.